### DISCUSSION TITLE
Hydrodynamic bearings

### DISCUSSION THREAD

**Discussion by nilswagner** [petrobras/ross#947] [Q&A]
Hi all,

Is it possible to compute the stiffness and damping parameters of hydrodynamic bearings for varying speeds by using ross?
The example is taken from the following paper

Model Reduction of Rotor-Foundation Systems Using the Approximate Invariant Manifold Method
https://doi.org/10.21203/rs.3.rs-2222988/v1

**Comment by ViniciusTxc3** [comment]
@nilswagner currently, there is the `THDCylindrical()` function that returns the stiffness and damping coefficients for hydrodynamic bearings, you can pass several speeds as in the example below:

```python
import numpy as np
from ross.fluid_flow.cylindrical import THDCylindrical
from ross.units import Q_  

_frequency = Q_([900], "RPM")
bearing = THDCylindrical(
        axial_length=0.263144,
        journal_radius=0.2,
        radial_clearance=1.95e-4,
        elements_circumferential=11,
        elements_axial=3,
        n_pad=2,
        pad_arc_length=176,
        preload=0,
        geometry="circular",
        reference_temperature=50,
        speed=_frequency,
        load_x_direction=0,
        load_y_direction=-112814.91,
        groove_factor=[0.52, 0.48],
        lubricant="ISOVG32",
        node=3,
        sommerfeld_type=2,
        initial_guess=[0.1, -0.1],
        method="perturbation",
        operating_type="flooded",
        injection_pressure=0,
        oil_flow=37.86,
        show_coef=False,
        print_result=False,
        print_progress=False,
        print_time=False,
    )
print(bearing)
```

**Reply by Pierre-zhu** [reply]
can we calculate bearing clearance close to 1.2e-3?  which for water lubrication.

**Reply by ViniciusTxc3** [reply]
Yes, you could pass the `radial_clearance` directly. However, you would need to insert water as a lubricant in the _ross > fluid_flow > lubricants.py_ . I will give an example below:

![image](https://github.com/user-attachments/assets/7a5e2040-d2b3-47b1-ad79-e51b54164f4b)

An example final code:

```python
    _frequency = Q_([900], "RPM")
    _radial_clarence = 1.2e-3

    bearing = THDCylindrical(
            axial_length=0.263144,
            journal_radius=0.2,
            radial_clearance=_radial_clarence,
            elements_circumferential=11,
            elements_axial=3,
            n_pad=2,
            pad_arc_length=176,
            preload=0,
            geometry="circular",
            reference_temperature=50,
            speed=_frequency,
            load_x_direction=0,
            load_y_direction=-112814.91,
            groove_factor=[0.52, 0.48],
            lubricant="WATER",
            node=3,
            sommerfeld_type=2,
            initial_guess=[0.1, -0.1],
            method="perturbation",
            operating_type="flooded",
            injection_pressure=0,
            oil_flow=37.86,
            show_coef=False,
            print_result=False,
            print_progress=False,
            print_time=False,
        )
    print(bearing)
```

**Reply by Pierre-zhu** [reply]
OK,thanks, I will try

**Reply by Pierre-zhu** [reply]
when I run my code, I get following error:

  File d:\ProgramData\Anaconda3\Lib\site-packages\ross\fluid_flow\cylindrical.py:237 in __init__
    lubricant_properties = lubricants_dict[self.lubricant]

KeyError: 'SEAWATER'

but  I already add seawater in D:\ProgramData\anaconda3\Lib\site-packages\ross\fluid_flow\lubricants.py

**Reply by ViniciusTxc3** [reply]
I believe you are running ross directly with pip. For my example above, you would have to clone our repository and insert it manually.

As a simpler solution, We're uploading an update where it will be possible to insert the lubricant directly as a dictionary in PR #1147 also adding seawater as a new lubricant.

Note: After this update, to import the THDCylindrical class, it will be as follows `from ross.bearing.cylindrical import THDCylindrical`.

This is an example code that I ran in my tests:

```python
import numpy as np
from ross.bearing.cylindrical import THDCylindrical
from ross.units import Q_  

_frequency = Q_([900], "RPM")
_radial_clarence = 1.2e-3
_lubricant_seawater = {
    "liquid_viscosity1": Q_(1.08e-03, "Pa*s").to_base_units().m,
    "temperature1": Q_(20, "degC").to_base_units().m,
    "liquid_viscosity2": Q_(0.35e-03, "Pa*s").to_base_units().m,
    "temperature2": Q_(100, "degC").to_base_units().m,
    "liquid_density": Q_(1025, "kg/m**3").to_base_units().m,
    "liquid_specific_heat": Q_(3850, "J/(kg*degK)").to_base_units().m,
    "liquid_thermal_conductivity": Q_(0.6, "W/(m*degC)").to_base_units().m,
    "vapour_viscosity": Q_(9.75e-06, "Pa*s").to_base_units().m,
    "vapour_density": Q_(0.597, "kg/m**3").to_base_units().m,
    "vapour_specific_heat": Q_(2010, "J/(kg*degC)").to_base_units().m,
    "vapour_thermal_conductivity": Q_(0.025, "W/(m*degK)").to_base_units().m,
    "saturation_pressure": Q_(101325, "Pa").to_base_units().m,
    "surface_tension": Q_(0.074, "N/m").to_base_units().m,
    "molecular_weight": Q_(18.015, "g/mol").to_base_units().m,
}

bearing = THDCylindrical(
        axial_length=0.263144,
        journal_radius=0.2,
        radial_clearance=_radial_clarence,
        elements_circumferential=11,
        elements_axial=3,
        n_pad=2,
        pad_arc_length=176,
        preload=0,
        geometry="circular",
        reference_temperature=50,
        speed=_frequency,
        load_x_direction=0,
        load_y_direction=-112814.91,
        groove_factor=[0.52, 0.48],
        lubricant=_lubricant_seawater,
        node=3,
        sommerfeld_type=2,
        initial_guess=[0.1, -0.1],
        method="perturbation",
        operating_type="flooded",
        injection_pressure=0,
        oil_flow=37.86,
        show_coef=False,
        print_result=False,
        print_progress=False,
        print_time=False,
    )
print(bearing)
```

**Reply by Pierre-zhu** [reply]
OK，thanks , I can use "seawater" now accoring to your guide.
But I try to calculation my example bearing , still some errors  as such bearing without arc , full circular. `bearing` be define as following:
`
bearingCase1 = THDCylindrical(
        axial_length=1.175,  # m 
        journal_radius=0.385/2.0, #m
        radial_clearance=1.29/2000.0, #m
        elements_circumferential=11,
        elements_axial=2, #Number of volumes along the Z direction (axial direction).
        n_pad=2,
        pad_arc_length=360, #Arc length of each pad. The unit is degree.
        preload=0,
        geometry="circular",
        reference_temperature=30,
        speed=Q_([153], "RPM"),  # rad/s
        load_x_direction=0, #N
        load_y_direction=-7.5e4, # N
        groove_factor=[0.0, 0.0],
        lubricant="SEAWATER",
        node=3,
        sommerfeld_type=2,
        initial_guess=[0.1, -0.1], #Array with eccentricity ratio and attitude angle
        method="perturbation",
        operating_type="flooded",
        injection_pressure=0,
        oil_flow=57.8, #litre/min
        show_coef=False,
        print_result=False,
        print_progress=False,
        print_time=False,
    )
print(bearingCase1 )

`


error as following:
d:\ProgramData\Anaconda3\Lib\site-packages\ross\bearing\cylindrical.py:1867: OptimizeWarning: Covariance of the parameters could not be estimated
  popt, pcov = curve_fit(viscosity, xdata, ydata, p0=(6.0, 1.0))

**Reply by ViniciusTxc3** [reply]
Analyzing your case, there are some considerations that should be made:

The element values ​​need to be larger so that an axis of this size can be discretized, suggest:

```python
elements_circumferential=20,
elements_axial=20,
```
There are 2 pads, the value of pad_arc_length needs to be recalculated:
```python
pad_arc_length=176,
```

The groove_factor needs to be 0.5:
```python
groove_factor=[0.5, 0.5],
```

The `load_y_direction` seems that with this lubricant the eccentricity is high. Maybe it can't support this load. I did some tests with values ​​between -0.5e4 and -0.5e3 and got some results that you can use as a starting point.

I suggest enabling the parameters below so you can track the progress:
```python
print_result=True,
print_progress=True,
```

**Reply by Pierre-zhu** [reply]
Thanks ,I will try again following.
